File: src/main/java/com/ject/studytrip/stamp/presentation/dto/response/CreateStampResponse.java
package com.ject.studytrip.stamp.presentation.dto.response;

import com.ject.studytrip.stamp.application.dto.StampInfo;
import io.swagger.v3.oas.annotations.media.Schema;

public record CreateStampResponse(@Schema(description = "스탬프 ID") Long stampId) {
    public static CreateStampResponse of(StampInfo info) {
        return new CreateStampResponse(info.stampId());
    }
}

File: src/main/kotlin/com/ject/studytrip/stamp/presentation/dto/response/CreateStampResponse.kt
package com.ject.studytrip.stamp.presentation.dto.response

import com.ject.studytrip.stamp.application.dto.StampInfo
import io.swagger.v3.oas.annotations.media.Schema

data class CreateStampResponse(
    @field:Schema(description = "스탬프 ID")
    val stampId: Long,
) {
    companion object {
        fun of(stampInfo: StampInfo): CreateStampResponse = CreateStampResponse(stampInfo.stampId)
    }
}
